left two warnings， deal these later。
```
/__w/stonedb/stonedb/storage/tianmu/mm/tcm/span.cpp:50:9: warning: 'void* memset(void*, int, size_t)' offset [0, 31] is out of the bounds [0, 0] [-Warray-bounds]
/__w/stonedb/stonedb/storage/tianmu/system/channel.cpp:51:50: warning: '%02d' directive writing between 2 and 11 bytes into a region of size between 0 and 17 [-Wformat-overflow=]
```

--- a/cmake/maintainer.cmake
+++ b/cmake/maintainer.cmake
@@ -166,9 +166,6 @@ IF(MY_COMPILER_IS_CLANG)
   # -Wzero-as-null-pointer-constant
 ENDIF()
 
-# TIANMU UPGRADE
-SET(MY_WARNING_FLAGS "${MY_WARNING_FLAGS} -Wno-unknown-pragmas")
-
 # Turn on Werror (warning => error) when using maintainer mode.
 IF(MYSQL_MAINTAINER_MODE)
   IF(MSVC)
@@ -178,11 +175,10 @@ IF(MYSQL_MAINTAINER_MODE)
     STRING_APPEND(CMAKE_MODULE_LINKER_FLAGS " /WX")
     STRING_APPEND(CMAKE_SHARED_LINKER_FLAGS " /WX")
   ENDIF()
-# TIANMU UPGRADE
-#  IF(MY_COMPILER_IS_GNU_OR_CLANG)
-#    STRING_APPEND(MY_C_WARNING_FLAGS   " -Werror")
-#    STRING_APPEND(MY_CXX_WARNING_FLAGS " -Werror")
-#  ENDIF()
+  IF(MY_COMPILER_IS_GNU_OR_CLANG)
+    STRING_APPEND(MY_C_WARNING_FLAGS   " -Werror")
+    STRING_APPEND(MY_CXX_WARNING_FLAGS " -Werror")
+  ENDIF()
 ENDIF()
 
 # Set warning flags for gcc/g++/clang/clang++

--- a/storage/tianmu/CMakeLists.txt
+++ b/storage/tianmu/CMakeLists.txt
@@ -93,5 +93,5 @@ TARGET_INCLUDE_DIRECTORIES(tianmu PRIVATE ${CMAKE_SOURCE_DIR}/storage/tianmu/bas
 TARGET_COMPILE_OPTIONS(tianmu PRIVATE -Wno-error -Wextra -Wall)
 TARGET_COMPILE_OPTIONS(tianmu PRIVATE -Wunused-parameter)
 TARGET_COMPILE_OPTIONS(tianmu PRIVATE -Wsuggest-override)
-
+TARGET_COMPILE_OPTIONS(tianmu PRIVATE -fno-strict-aliasing)
 add_subdirectory(async_tests)

--- a/storage/tianmu/base/core/future.h
+++ b/storage/tianmu/base/core/future.h
@@ -726,6 +726,7 @@ class future {
       }
     }
     assert(0 && "we should not be here");
+    return make_ready_future<T...>();  // just swallow warning "-Wreturn-type"
   }
 
   template <typename... U>
@@ -959,6 +960,7 @@ class future {
             return result.rethrow_with_nested();
           }
           assert(0 && "we should not be here");
+          return result.rethrow_with_nested();  // just swallow warning "-Wreturn-type"
         }
       });
     }

--- a/storage/tianmu/base/core/posix.h
+++ b/storage/tianmu/base/core/posix.h
@@ -448,7 +448,7 @@ inline void pin_this_thread(unsigned cpu_id) {
   cpu_set_t cs;
   CPU_ZERO(&cs);
   CPU_SET(cpu_id, &cs);
-  auto r = pthread_setaffinity_np(pthread_self(), sizeof(cs), &cs);
+  auto r [[maybe_unused]] = pthread_setaffinity_np(pthread_self(), sizeof(cs), &cs);
   assert(r == 0);
 }
 

--- a/storage/tianmu/base/core/reactor.cpp
+++ b/storage/tianmu/base/core/reactor.cpp
@@ -435,7 +435,7 @@ future<> reactor_backend_epoll::get_epoll_future(pollable_fd_state &pfd, promise
     ::epoll_event eevt;
     eevt.events = pfd.events_epoll;
     eevt.data.ptr = &pfd;
-    int r = ::epoll_ctl(_epollfd.get(), ctl, pfd.fd.get(), &eevt);
+    int r [[maybe_unused]] = ::epoll_ctl(_epollfd.get(), ctl, pfd.fd.get(), &eevt);
     assert(r == 0);
     engine().start_epoll();
   }
@@ -451,7 +451,7 @@ void reactor_backend_epoll::abort_fd(pollable_fd_state &pfd, std::exception_ptr 
     ::epoll_event eevt;
     eevt.events = pfd.events_epoll;
     eevt.data.ptr = &pfd;
-    int r = ::epoll_ctl(_epollfd.get(), ctl, pfd.fd.get(), &eevt);
+    int r [[maybe_unused]] = ::epoll_ctl(_epollfd.get(), ctl, pfd.fd.get(), &eevt);
     assert(r == 0);
   }
   if (pfd.events_requested & event) {
@@ -1845,7 +1845,7 @@ file_desc writeable_eventfd::try_create_eventfd(size_t initial) {
 
 void writeable_eventfd::signal(size_t count) {
   uint64_t c = count;
-  auto r = _fd.write(&c, sizeof(c));
+  auto r [[maybe_unused]] = _fd.write(&c, sizeof(c));
   assert(r == sizeof(c));
 }
 
@@ -1859,7 +1859,7 @@ file_desc readable_eventfd::try_create_eventfd(size_t initial) {
 future<size_t> readable_eventfd::wait() {
   return engine().readable(*_fd._s).then([this] {
     uint64_t count;
-    int r = ::read(_fd.get_fd(), &count, sizeof(count));
+    int r [[maybe_unused]] = ::read(_fd.get_fd(), &count, sizeof(count));
     assert(r == sizeof(count));
     return make_ready_future<size_t>(count);
   });
@@ -1963,7 +1963,7 @@ void smp::allocate_reactor(unsigned id) {
   // we cannot just write "local_engin = new reactor" since reactor's
   // constructor uses local_engine
   void *buf;
-  int r = posix_memalign(&buf, cache_line_size, sizeof(reactor));
+  int r [[maybe_unused]] = posix_memalign(&buf, cache_line_size, sizeof(reactor));
   assert(r == 0);
   local_engine = reinterpret_cast<reactor *>(buf);
   new (buf) reactor(id);

--- a/storage/tianmu/base/core/systemwide_memory_barrier.cpp
+++ b/storage/tianmu/base/core/systemwide_memory_barrier.cpp
@@ -37,7 +37,7 @@ void systemwide_memory_barrier() {
     assert(mem != MAP_FAILED);
     return reinterpret_cast<char *>(mem);
   }();
-  int r1 = mprotect(mem, getpagesize(), PROT_READ | PROT_WRITE);
+  int r1 [[maybe_unused]] = mprotect(mem, getpagesize(), PROT_READ | PROT_WRITE);
   assert(r1 == 0);
   // Force page into memory to avoid next mprotect() attempting to be clever
   *mem = 3;
@@ -45,7 +45,7 @@ void systemwide_memory_barrier() {
   // lower permissions to force kernel to send IPI to all threads, with
   // a side effect of executing a memory barrier on those threads
   // FIXME: does this work on ARM?
-  int r2 = mprotect(mem, getpagesize(), PROT_READ);
+  int r2 [[maybe_unused]] = mprotect(mem, getpagesize(), PROT_READ);
   assert(r2 == 0);
 }
 

--- a/storage/tianmu/core/temp_table_low.cpp
+++ b/storage/tianmu/core/temp_table_low.cpp
@@ -564,11 +564,11 @@ void TempTable::VerifyAttrsSizes()  // verifies attr[i].field_size basing on the
       if (attr->term_.vc->MaxStringSize() < STRING_LENGTH_THRESHOLD) {
         attr->OverrideStringSize(attr->term_.vc->MaxStringSize());
       } else {
-        vcolumn::VirtualColumn *vc = attr->term_.vc;
         int max_length = attr->term_.vc->MaxStringSize();
 // TODO, the code has some bug, max_length in some case wil be negative, see #671
 // comment the optimization code for temp solution;
 #if 0
+        vcolumn::VirtualColumn *vc = attrs[i]->term.vc;
         if (dynamic_cast<vcolumn::ExpressionColumn *>(vc)) {
           auto &var_map = dynamic_cast<vcolumn::ExpressionColumn *>(vc)->GetVarMap();
           for (auto &it : var_map) {

--- a/storage/tianmu/core/transaction.cpp
+++ b/storage/tianmu/core/transaction.cpp
@@ -77,7 +77,7 @@ void Transaction::AddTableWRIfNeeded(std::shared_ptr<TableShare> &share) {
 }
 
 void Transaction::Commit([[maybe_unused]] THD *thd) {
-  TIANMU_LOG(LogCtl_Level::INFO, "txn commit, modified tables size in txn: [%d].", m_modified_tables.size());
+  TIANMU_LOG(LogCtl_Level::INFO, "txn commit, modified tables size in txn: [%lu].", m_modified_tables.size());
   for (auto const &iter : m_modified_tables) iter.second->CommitVersion();
 
   m_modified_tables.clear();

--- a/storage/tianmu/handler/ha_my_tianmu.cpp
+++ b/storage/tianmu/handler/ha_my_tianmu.cpp
@@ -47,25 +47,6 @@ static bool ForbiddenMySQLQueryPath([[maybe_unused]] Query_expression *qe) {
   return (tianmu_sysvar_allowmysqlquerypath == 0);
 }
 
-static bool Tianmu_SetStatementAllowed(THD *thd, Query_expression *qe) {
-  if (AtLeastOneTianmuTableInvolved(qe)) {
-    if (ForbiddenMySQLQueryPath(qe)) {
-      my_message(static_cast<int>(common::ErrorCode::UNKNOWN_ERROR),
-                 "Queries inside SET statements are not supported. "
-                 "Enable the MySQL core::Query Path in my.cnf to execute the query "
-                 "with reduced "
-                 "performance.",
-                 MYF(0));
-      return false;
-    } else
-      push_warning(thd, Sql_condition::SL_NOTE, ER_UNKNOWN_ERROR,
-                   "SET statement not supported by the Tianmu Optimizer. The "
-                   "query executed "
-                   "by MySQL engine.");
-  }
-  return true;
-}
-
 void ha_my_tianmu_update_and_store_col_comment(TABLE *table, int field_id, Field *source_field, int source_field_id,
                                                CHARSET_INFO *cs) {
   try {

